src/com/google/InstrumentationMapping.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2020 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.ProductionInstrumentationReporter.InstrumentationType;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * The class contains the data which is provided by the Production Instrumentation pass in the
+ * closure compiler. It will read the data from the file name created with the
+ * --instrument_mapping_report and is required to follow the format used by Closure Compiler class
+ * VariableMap.
+ */
+public class InstrumentationMapping {
```

Comment:
Can this be package-private & final, so it's only used by ProductionInstrumentationReporter?

``` java
final class InstrumentationMapping {
```